devlib/utils/misc.py

```diff
@@ -590,6 +591,31 @@ def __str__(self):
         return message.format(self.filepath, self.lineno, self.message)
 
 
+def load_struct_from_yaml(filepath):
+    '''
+    Parses a config structure from a YAML file.
+    The structure should be composed of basic Python types.
+
+    Args:
+        filepath (str): Input file which contains YAML data.
+
+    Raises:
+        LoadSyntaxError: if there is a syntax error in YAML data.
+
+    Returns:
+        dict: which contains parsed YAML data.
```

Comment:
I would recommend using the autodoc style as this is the most native one with Sphinx (you can see what we use in LISA)